tests/unit-tests/data-port/test-class-sensei-data-port-utilites.php

```diff
@@ -37,4 +37,181 @@ public function testUserIsUpdatedIfExists() {
+		$taxonomy_name = 'course-category';
+
+		$term_a = Sensei_Data_Port_Utilities::get_term( implode( ' > ', $term_path_a ), $taxonomy_name );
+		$term_b = Sensei_Data_Port_Utilities::get_term( implode( ' > ', $term_path_b ), $taxonomy_name );
+
+		$this->assertTermPathValid( $term_path_a, $term_a, $taxonomy_name );
+		$this->assertTermPathValid( $term_path_b, $term_b, $taxonomy_name );
+
+		$term_path_a_ids = [];
+		while ( $term_a ) {
+			$term_path_a_ids[] = $term_a->term_id;
+
+			if ( ! $term_a->parent ) {
+				break;
+			}
+
+			$term_a = get_term_by( 'id', $term_a->parent, $taxonomy_name );
+		}
+
+		$term_path_b_ids = [];
+		while ( $term_b ) {
+			$term_path_b_ids[] = $term_b->term_id;
+
+			if ( ! $term_b->parent ) {
+				break;
+			}
+
+			$term_b = get_term_by( 'id', $term_b->parent, $taxonomy_name );
+		}
+
+		$this->assertEquals( count( $term_path_a ), count( $term_path_a_ids ), 'A: IDs should match size of path' );
+		$this->assertEquals( count( $term_path_b ), count( $term_path_b_ids ), 'B: IDs should match size of path' );
+		$this->assertEquals( 1, count( array_intersect( $term_path_a_ids, $term_path_b_ids ) ), 'There should be no similar IDs in the paths' );
+		$this->assertEquals( $term_path_a_ids[2], $term_path_b_ids[1], 'The first term in each path should be the same' );
+	}
+
+	/**
+	 * Tests a term path on a non-hierarchical taxonomy.
+	 */
+	public function testGetTermNonHierarchicalPath() {
+		$term_path     = [ 'Just > A > Sneaky > Dinosaur' ];
+		$taxonomy_name = 'lesson-tag';
+
+		$term = Sensei_Data_Port_Utilities::get_term( implode( ' > ', $term_path ), $taxonomy_name );
+		$this->assertTermPathValid( $term_path, $term, $taxonomy_name );
+	}
+
+	/**
+	 * Tests a term path for modules with a teacher.
+	 */
+	public function testGetTermModulePathTeacher() {
+		$teacher_id      = $this->factory->user->create( array( 'role' => 'teacher' ) );
+		$term_path       = [ 'Module A', 'Module A.1', 'Module A.2', 'Module A.3' ];
+		$term_path_slugs = [ $teacher_id . '-module-a', $teacher_id . '-module-a-1', $teacher_id . '-module-a-2', $teacher_id . '-module-a-3' ];
+
+		$taxonomy_name = 'module';
+
+		$term = Sensei_Data_Port_Utilities::get_term( implode( ' > ', $term_path ), $taxonomy_name, $teacher_id );
+		$this->assertTermPathValid( $term_path_slugs, $term, $taxonomy_name, 'slug' );
+	}
+
+	/**
+	 * Tests a term path for modules with a teacher.
+	 */
+	public function testGetTermModulePathAdmin() {
+		$admin_id        = $this->factory->user->create( array( 'role' => 'administrator' ) );
+		$term_path       = [ 'Module A', 'Module A.1', 'Module A.2', 'Module A.3' ];
+		$term_path_slugs = [ 'module-a', 'module-a-1', 'module-a-2', 'module-a-3' ];
+
+		$taxonomy_name = 'module';
+
+		$term = Sensei_Data_Port_Utilities::get_term( implode( ' > ', $term_path ), $taxonomy_name, $admin_id );
+		$this->assertTermPathValid( $term_path_slugs, $term, $taxonomy_name, 'slug' );
+	}
+
+	/**
+	 * Assert a term path is valid.
```

Comment:
Can we document what this method does?

Reply:
ce2c8873a34543b4d30adf386058758797e24804